hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/DeleteMetadataActor.scala

```diff
@@ -2,29 +2,38 @@ package cromwell.services.metadata.hybridcarbonite
 
 import java.time.OffsetDateTime
 
-import akka.actor.{Actor, ActorLogging, Props}
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import cromwell.core.WorkflowId
 import cromwell.services.MetadataServicesStore
+import cromwell.services.instrumentation.CromwellInstrumentation
 import cromwell.services.metadata.MetadataArchiveStatus
 import cromwell.services.metadata.MetadataArchiveStatus._
-import cromwell.services.metadata.hybridcarbonite.DeleteMetadataActor.DeleteMetadataAction
+import cromwell.services.metadata.hybridcarbonite.DeleteMetadataActor._
+import cromwell.services.metadata.hybridcarbonite.WorkflowsToDeleteMetadataMetricHelperActor.{CalculateMetricAndSend, SendPrecalculatedMetric}
 import cromwell.services.metadata.impl.MetadataDatabaseAccess
 
 import scala.util.{Failure, Success}
 import scala.concurrent.duration._
 
-class DeleteMetadataActor(metadataDeletionConfig: MetadataDeletionConfig) extends Actor
+class DeleteMetadataActor(metadataDeletionConfig: MetadataDeletionConfig, override val serviceRegistryActor: ActorRef) extends Actor
   with ActorLogging
   with MetadataDatabaseAccess
-  with MetadataServicesStore {
+  with MetadataServicesStore
+  with CromwellInstrumentation {
 
   implicit val ec = context.dispatcher
 
   metadataDeletionConfig.intervalOpt.map { interval =>
     context.system.scheduler.schedule(5.minutes, interval, self, DeleteMetadataAction)
   }
 
-  override def receive: Receive = {
+  val workflowsToDeleteMetadataMetricHelperActor =
+    context.actorOf(WorkflowsToDeleteMetadataMetricHelperActor.props(self, serviceRegistryActor))
+
+  override def receive: Receive = receive(isMetricActorFree = true)
```

Comment:
That was my initial implementation, but Idea suggested to replace it with `become` behavior. And actually it looks more in the spirit of Akka than var.

Reply:
hmm, I guess I disagree with Idea in this case, but I won't push it! I suspect the most "akka"y way to do this would be to have two separately coded methods to switch between, rather than methods which get generated ad hoc using a boolean parameter. But there's so much shared between them that that seems like huge overkill here.

TL;DR: You should stick with whichever you prefer!